### PR TITLE
Migrating tests of GitAPITestCase from JUnit 3 to JUnit 4

### PR DESCRIPTION
**Pull request by GAyan17**
## [JENKINS-60940](https://issues.jenkins-ci.org/browse/JENKINS-60940) - Migrating tests of GitAPITestCase from JUnit 3 to JUnit 4

In this PR, the test migrated is:
- testCleanWithParameters
- testClean
- testHasGitRepo

Describe the big picture of your changes here to explain to the maintainers why this pull request should be accepted.
If it fixes a bug or resolves a feature request, include a link to the issue.

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/jenkinsci/git-client-plugin/blob/master/CONTRIBUTING.adoc) doc
- [x] I have referenced the Jira issue related to my changes in one or more commit messages
- [x] I have added tests that verify my changes
- [x] Unit tests pass locally with my changes
- [x] I have added documentation as necessary
- [x] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes
- [x] I have interactively tested my changes

## Types of changes

What types of changes does your code introduce?

- [x] Tests

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/gitclient/WorkspaceWithRepo.java [hunk 1]
```diff
@@ -72,7 +73,7 @@ private void renameAndDeleteDir(Path srcDir, String destDirName) {
      * Returns path to the local mirror directory.
      *
      * @return path to the local mirrror directory
-     * @throws IOException on I/O error
+     * @throws IOException          on I/O error
```
Comment (MarkEWaite) [inline src/test/java/org/jenkinsci/plugins/gitclient/WorkspaceWithRepo.java:76]: Please don't make whiitespace changes to existing files.  It creates unnecessary barriers to other pull requests.
```suggestion
     * @throws IOException on I/O error
```

File: src/test/java/org/jenkinsci/plugins/gitclient/WorkspaceWithRepo.java [hunk 2]
```diff
@@ -39,7 +39,8 @@ private void createEnv(File gitDir, String gitImplName, TaskListener listener) t
         cliGitCommand = new CliGitCommand(gitClient);
     }
 
-    private GitClient setupGitClient(File gitFileDir, @NonNull String gitImplName, TaskListener listener) throws Exception {
+    private GitClient setupGitClient(File gitFileDir, @NonNull String gitImplName, TaskListener listener)
+            throws Exception {
```
Comment (MarkEWaite) [inline src/test/java/org/jenkinsci/plugins/gitclient/WorkspaceWithRepo.java:43]: Please don't make white space changes in existing files.  It creates barriers for other pull requests.
```suggestion
    private GitClient setupGitClient(File gitFileDir, @NonNull String gitImplName, TaskListener listener) throws Exception {
```

File: src/test/java/org/jenkinsci/plugins/gitclient/WorkspaceWithRepo.java [hunk 3]
```diff
@@ -92,7 +93,8 @@ String localMirror() throws IOException, InterruptedException {
                      * the final destination directory.
                      */
                     Path tempClonePath = Files.createTempDirectory(targetDir.toPath(), "clone-");
-                    cliGitCommand.run("clone", "--reference", f.getCanonicalPath(), "--mirror", repoURL, tempClonePath.toFile().getAbsolutePath());
+                    cliGitCommand.run("clone", "--reference", f.getCanonicalPath(), "--mirror", repoURL,
+                            tempClonePath.toFile().getAbsolutePath());
```
Comment (MarkEWaite) [inline src/test/java/org/jenkinsci/plugins/gitclient/WorkspaceWithRepo.java:97]: Please remove the pure white space changes in the existing files.
```suggestion
                    cliGitCommand.run("clone", "--reference", f.getCanonicalPath(), "--mirror", repoURL, tempClonePath.toFile().getAbsolutePath());
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Needs to have the changes reverted that are purely white space.  White space changes make other pull requests have unnecessary conflicts.  See the CONTRIBUTING file for more details describing why this plugin needs to leave its ugly formatting alone.

**Comment by GAyan17** [comment]
Thanks @MarkEWaite for the code suggestion review, I will surely keep that in mind for other pull requests.

**Review by MarkEWaite — APPROVED** [review]

**Pull request merged**